packages/app-core/src/alerting/fifo.ts

```diff
@@ -0,0 +1,63 @@
+
+export class RingFifo<T> {
+	protected store: T[];
+	protected head = 0;
+	protected tail = 0;
+	protected full = false;
+
+	constructor(size: number) {
+		this.store = Array.from({length: size});
+	}
+
+	get size() {
```

Comment:
nit - `size` used to both describe the total capacity and the current state of how many elements are stored

Reply:
Discussed offline, `capacity` for the constructor, `count` will replace this method